src/uct/ib/dc/verbs/dc_verbs.c

```diff
@@ -543,38 +551,53 @@ ucs_status_t uct_dc_verbs_iface_flush(uct_iface_h tl_iface, unsigned flags, uct_
     if (comp != NULL) {
         return UCS_ERR_UNSUPPORTED;
     }
+
     status = uct_dc_verbs_flush_dcis(iface);
     if (status == UCS_OK) {
         UCT_TL_IFACE_STAT_FLUSH(&iface->super.super.super);
-    } 
-    else if (status == UCS_INPROGRESS) {
+    } else if (status == UCS_INPROGRESS) {
         UCT_TL_IFACE_STAT_FLUSH_WAIT(&iface->super.super.super);
     }
     return status;
 }
 
 ucs_status_t uct_dc_verbs_ep_flush(uct_ep_h tl_ep, unsigned flags, uct_completion_t *comp)
 {
```

Comment:
need to check that ep->state != UCT_DC_EP_TX_WAIT in uct_dc_ep_pending_add() and make sure that flush uses same logic.